### PR TITLE
New dataset, add prompt formatting to dataset.py

### PR DESCRIPTION
**Pull request by dagrayvid**
This PR includes the following changes:
- Add code for generating the dataset from openorca
  - New dataset removes the llama prompt format / system prompt
  - New dataset adds index to dataset
- Add prompt formatting to dataset.py including some known prompt (and system_prompt) formats
- Add option to control max sequence length included in dataset

### CODE REVIEW

File: plugins/tgis_grpc_plugin.py [hunk 1]
```diff
@@ -61,15 +61,22 @@ def make_request(self, query: dict, user_id: int):
             params=generation_pb2_grpc.generation__pb2.Parameters(
                 method=generation_pb2_grpc.generation__pb2.GREEDY,
                 stopping=generation_pb2_grpc.generation__pb2.StoppingCriteria(
-                    max_new_tokens=query["max_new_tokens"],
-                    min_new_tokens=query["min_new_tokens"],
+                    max_new_tokens=query["output_tokens"],
+                    min_new_tokens=query["output_tokens"],
```
Comment (npalaska) [inline plugins/tgis_grpc_plugin.py:65]: Hmm, so we are forcing the model to respond with an exact number of tokens. Probably good for the benchmarking exercises I guess.

**Review by npalaska — APPROVED** [review]
I think this looks really good 👍

File: plugins/tgis_grpc_plugin.py (hunk 1 above)
Comment (dagrayvid) [reply plugins/tgis_grpc_plugin.py:65]: This is actually how it was before, but in the dataset we had two fields per element (min_new_tokens, max_new_tokens) but they were always the same

File: dataset.py [hunk 2]
```diff
@@ -26,34 +27,71 @@ def get_next_n_queries(self, n):
...
                 json_object = json.loads(line.strip())
             except json.JSONDecodeError as e:
                 logging.error("Error decoding JSON in file %s %s", filename, e)
                 continue
             try:
-                input_tokens = json_object["tok_input_length"]
-                input_data = json_object["input"]
-                output_tokens = input_data["min_new_tokens"]
+                input_tokens = int(json_object["tok_input_length"])
+                output_tokens = int(json_object["tok_output_length"])
```
Comment (kpouget) [inline dataset.py:56]: you don't really need to `int()` cast here, json gives it to you in the right format
(except if some datasets are inconsistent ...)

File: dataset.py [hunk 3]
```diff
@@ -26,34 +27,71 @@ def get_next_n_queries(self, n):
...
+        # [1:] to skip the first line, it contains metadata
+        lines = file.readlines()[1:]
+        random.Random(dataset_seed).shuffle(lines)
+
+        for line in lines:
             # Load each line as a JSON object
             try:
                 json_object = json.loads(line.strip())
             except json.JSONDecodeError as e:
                 logging.error("Error decoding JSON in file %s %s", filename, e)
```
Comment (kpouget) [inline dataset.py:52]: you may want to add a line indicator here, but can be in another PR

File: dataset.py [hunk 4]
```diff
@@ -26,34 +27,71 @@ def get_next_n_queries(self, n):
...
+            if (output_tokens < max_output_tokens
+                and input_tokens < max_input_tokens
+                and input_tokens > min_input_tokens
+                and sequence_tokens < max_sequence_tokens):
+                input_data = {
+                    "text": prompt_format.format(prompt=prompt, 
+                                                  system_prompt=system_prompt),
+                    "input_id": input_id,
+                    "input_tokens": input_tokens,
+                    "output_tokens": output_tokens,
```
Comment (kpouget) [inline dataset.py:76]: can you store the `stopReason`?
```
2024-03-05 09:17:25,842 p=2468 u=psap-ci-runner n=ansible | <stdout> {
2024-03-05 09:17:25,842 p=2468 u=psap-ci-runner n=ansible | <stdout>   "responses": [
2024-03-05 09:17:25,842 p=2468 u=psap-ci-runner n=ansible | <stdout>     {
2024-03-05 09:17:25,842 p=2468 u=psap-ci-runner n=ansible | <stdout>       "generatedTokenCount": 17,
2024-03-05 09:17:25,842 p=2468 u=psap-ci-runner n=ansible | <stdout>       "text": "74 degrees F.C., a temperature of 74 degrees F",
2024-03-05 09:17:25,842 p=2468 u=psap-ci-runner n=ansible | <stdout>       "inputTokenCount": 9,
2024-03-05 09:17:25,842 p=2468 u=psap-ci-runner n=ansible | <stdout>       "stopReason": "EOS_TOKEN"
2024-03-05 09:17:25,842 p=2468 u=psap-ci-runner n=ansible | <stdout>     }
2024-03-05 09:17:25,842 p=2468 u=psap-ci-runner n=ansible | <stdout>   ]
2024-03-05 09:17:25,842 p=2468 u=psap-ci-runner n=ansible | <stdout> }
```

**Review by kpouget — COMMENTED** [review]
thanks @dagrayvid , looks good from what I can tell 👍🏻

**Review by npalaska — APPROVED** [review]

File: load_test.py [hunk 5]
```diff
@@ -141,63 +141,70 @@ def main(args):
...
+                stop_q=stop_q,
+                results_pipe=send_results,
+                plugin=plugin,
+                logger_q=logger_q,
+                log_level=args.log_level,
+            )
+            proc = mp_ctx.Process(target=user.run_user_process)
+            procs.append(proc)
+            logging.info("Starting %s", proc)
+            proc.start()
```
Comment (npalaska) [inline load_test.py:176]: I feel like we also need similar error handling in the graceful_shutdown function. For example when we `.join` all the user processes we don't want to hang there trying to terminate. We have to wait for all the processes to terminate on its own but If one of the process is taking forever its going to block all the other termination and again resulting in main process hanging.

Comment (dagrayvid) [reply load_test.py:176]: I think I agree, but let's address it in a follow-up
